django/thunderstore/repository/models/service_account.py

```diff
@@ -0,0 +1,18 @@
+import ulid2
+from django.conf import settings
+from django.db import models
+
+
+class ServiceAccountMetadata(models.Model):
+    uuid = models.UUIDField(default=ulid2.generate_ulid_as_uuid, primary_key=True)
+    user = models.OneToOneField(
+        settings.AUTH_USER_MODEL,
+        related_name="service_account_metadata",
+        on_delete=models.CASCADE,
+    )
+    is_service_account = models.BooleanField(default=False)
+    owner = models.ForeignKey(
+        "repository.UploaderIdentity",
+        related_name="service_accounts",
+        on_delete=models.CASCADE,
+    )
```

Comment:
Unsure about the need for this field. There's 2 main approaches I can think of:

## Every user has a `ServiceAccountMetadata`
When a user registers their `ServiceAccountMetadata` would be created using signals. This would mean `is_service_account` is needed and `owner` would need to become optional.

## Service accounts have `ServiceAccountMetadata`
If a user has a `service_account_metadata`, it's a service account. If not, they're a regular user. This would make `is_service_account` useless.

Reply:
2 is better for separation of responsibilities, but 1 is easier to use in code (since normally Django just throws an error at you if you try to refer to a non-existing OneToOne relation IIRC)

I would say let's go with option 2, as that should isolate the service account feature more nicely and make accidental pollution of it with non-service-account related metadata more difficult. It is a bit less nice to query against however if you can't expect every user to contain that metadata.

Reply:
I had a feeling it would raise but I didn't end up checking. I guess this means renaming to `ServiceAccount` makes more sense although there might be another name which captures that it isn't quite an *account* that it is representing but instead information about the *parent* being one.

Reply:
That really depends on how you view the terms, given that a `ServiceAccount` isn't equivalent to an `User` or isn't even supposed to be, it just instruments an user behind the scense. Kind of like how `cronjobs` are just small scripts that instrument a user, but you still call them cronjobs rather than `cronjobmetadata` or whatever

Reply:
Let's put it this way, even if implementation-wise `ServiceAccount` piggybacks off of the normal `User` model, the way we expose it outside and to the users is akin to them being an alternative to the normal `User` model. Additionally, if at some point we want to remove the `User` model usage from the `ServiceAccount` model entirely, we can do that behind the scenes, since our primary interface for operating service account's should already be the `ServiceAccount` model itself.